src/carbonio-files-ui-common/views/components/sharing/publicLink/PublicLink.test.tsx

```diff
@@ -162,145 +264,709 @@ describe.each<Node['__typename']>(['File', 'Folder'])('Public Link', (nodeType)
+					firstOfNextMonth.getMonth(),
+					firstOfNextMonth.getDate(),
+					23,
+					59
+				),
+				'DD/MM/YYYY'
+			);
+			expect(screen.getByText(expiresOnDate)).toBeVisible();
+		});
+	});
+
+	describe('Edit button', () => {
+		it('should render the empty input fields when click on edit button', async () => {
+			const node = populateNode(nodeType);
+			const linkTitle = 'Link title';
+			const link = populateLink(node);
+			link.expires_at = null;
+			link.description = '';
+			const mocks = {
+				Query: {
+					getLinks: mockGetLinks([])
+				},
+				Mutation: {
+					createLink: mockCreateLink(link)
+				}
+			} satisfies Partial<Resolvers>;
+			const linkName = 'Link name';
+			const { user } = setup(
+				<PublicLink
+					nodeId={node.id}
+					nodeName={node.name}
+					linkTitle={linkTitle}
+					linkName={linkName}
+					linkDescription={'Link description'}
+				/>,
+				{ mocks }
+			);
+			await user.click(screen.getByRole('button', { name: /add link/i }));
+			await user.click(screen.getByRole('button', { name: /generate link/i }));
+			await user.click(screen.getByRole('button', { name: /edit/i }));
+			const inputDescription = screen.getByRole('textbox', { name: /link's description/i });
+			expect(inputDescription).toBeVisible();
+			expect(inputDescription).toHaveValue('');
+			expect(screen.getByRole('textbox', { name: /expiration date/i })).toBeVisible();
+		});
+
+		it('should render the populated description and expiration date input fields when click on edit button', async () => {
+			const node = populateNode(nodeType);
+			const linkTitle = 'Link title';
+			const link = populateLink(node);
+			const firstOfNextMonth = getFirstOfNextMonth();
+			const expiresAt = initExpirationDate(firstOfNextMonth) as Date;
+			link.expires_at = expiresAt?.getTime();
+			link.description = 'This is the description';
+			const mocks = {
+				Query: {
+					getLinks: mockGetLinks([link])
+				}
+			} satisfies Partial<Resolvers>;
+			const { user } = setup(
+				<PublicLink
+					nodeId={node.id}
+					nodeName={node.name}
+					linkTitle={linkTitle}
+					linkName={'Link name'}
+					linkDescription={'Link description'}
+				/>,
+				{ mocks }
+			);
+			await screen.findByText(link.url as string);
+			expect(screen.getByText(link.description)).toBeVisible();
+			await user.click(screen.getByRole('button', { name: /edit/i }));
+			const inputDescription = screen.getByRole('textbox', { name: /link's description/i });
+			expect(inputDescription).toBeVisible();
+			expect(inputDescription).toHaveValue(link.description);
+			const expiresOnDate = formatDate(
+				new Date(
+					firstOfNextMonth.getFullYear(),
+					firstOfNextMonth.getMonth(),
+					firstOfNextMonth.getDate(),
+					23,
+					59
+				),
+				'DD/MM/YYYY'
+			);
+			expect(screen.getByRole('textbox', { name: /expiration date/i })).toBeVisible();
+			expect(screen.getByText(expiresOnDate)).toBeVisible();
+		});
+	});
+
+	describe('Modal', () => {
```

Comment:
I tried the second option, but it fails because each describe passes on both `it` cases.
For example:
- the remove button expects to have the `ICON_REGEXP.deletePermanently` icon, which is correct, but expects also that
- the remove button has the `ICON_REGEXP.revoke` icon, which is wrong, and vice versa.

So I opted for the first one.